jujugui/static/gui/src/app/app.js

```diff
@@ -832,10 +838,15 @@ YUI.add('juju-gui', function(Y) {
         const creds = this.controllerAPI.getCredentials();
         const gisf = this.get('gisf');
         const currentState = this.state.current;
+        const rootState = currentState ? currentState.root : null;
         // If an anonymous GISF user lands on the GUI at /new then don't
         // attempt to log into the controller.
-        if (!creds.areAvailable && gisf &&
-            (currentState && currentState.root === 'new')) {
+        if ((
+          !creds.areAvailable && gisf && rootState === 'new'
+        ) || (
+          this.anonymousMode && rootState !== 'login'
+        )) {
+          this.anonymousMode = true;
```

Comment:
Assuming lint doesn't complain, I might indent this block one degree more--right now it lines up with the conditionals which took me a moment to parse out.